Support S3 Compatible Blob Storage Providers

enisn opened the pull request:
Adds `ServiceURL` and `DisablePayloadSigning` configuration to the AWS Blob provider so it works with S3-compatible services (Cloudflare R2, MinIO, Backblaze B2, Wasabi, DigitalOcean Spaces).

`DisablePayloadSigning = true` makes the SDK send `UNSIGNED-PAYLOAD` instead of the streaming chunked signature that AWS SDK v4 uses by default. Required for Cloudflare R2 (rejects streaming signing with `501 NotImplemented`); leave `false` for AWS S3 and MinIO.

Test module reads `ServiceURL` / `DisablePayloadSigning` / `ContainerName` / `CreateContainerIfNotExists` from user secrets, so the existing `BlobContainer_Tests` base can target any S3-compatible service. When `ContainerName` is supplied the test prefixes every object with `abp-aws-test-run-{guid}/` so cleanup deletes only what this run created.

Locally tested end-to-end against a real Cloudflare R2 bucket and a Docker MinIO instance — both pass the full Save / Get / Delete / Exists / multi-tenant integration suite.

Closes #25503.

enisn committed:
Add ServiceURL configuration for S3-compatible APIs

- Introduced a new property `ServiceURL` in `AwsBlobProviderConfiguration` to allow custom service URLs for S3-compatible APIs like MinIO and DigitalOcean Spaces.
- Updated `AwsBlobProviderConfigurationNames` to include the new `ServiceURL` constant.
- Modified `DefaultAmazonS3ClientFactory` to utilize the `ServiceURL` when creating the S3 client configuration, enabling support for custom endpoints.

enisn committed:
Add unit tests for AwsBlobProviderConfiguration and DefaultAmazonS3ClientFactory

- Created `AwsBlobProviderConfiguration_Tests` to validate ServiceURL setting and retrieval.
- Implemented `DefaultAmazonS3ClientFactory_Tests` to ensure S3 client creation with and without custom ServiceURL.
- Tests cover scenarios for both S3-compatible services and default AWS S3 behavior.

enisn committed:
Enhance AWS Blob Storing Documentation for S3-Compatible Services

- Updated the AWS provider documentation to clarify support for S3-compatible storage services, including MinIO, DigitalOcean Spaces, and others.
- Added details on configuring the `ServiceURL` property for S3-compatible APIs.
- Included example configurations for MinIO, DigitalOcean Spaces, and Wasabi to assist users in setting up their environments.

enisn committed:
Add CloudFlare R2 example documentation

enisn committed:
Make region optional for some S3 compatible APIs

enisn committed:
Configure `ResponseChecksumValidation` hwne third party S3 providers used

enisn commented:
AWS SDK v4.0 is not compatible with 3rd party S3 APIs yet due to Payload Signing and SignCheck. (`STREAMING-AWS4-HMAC-SHA256-PAYLOAD`)

maliming commented on framework/src/Volo.Abp.BlobStoring.Aws/Volo/Abp/BlobStoring/Aws/DefaultAmazonS3ClientFactory.cs:
```diff
@@ -30,31 +30,58 @@ public DefaultAmazonS3ClientFactory(
+        return new AmazonS3Client(configuration.AccessKeyId, configuration.SecretAccessKey, clientConfig);
+    }
+
+    protected virtual AmazonS3Config CreateS3ClientConfig(AwsBlobProviderConfiguration configuration, RegionEndpoint? region)
```
Using the `Async` method will be better.

maliming committed:
Support Cloudflare R2 in AWS BLOB provider

- Add DisablePayloadSigning configuration to send UNSIGNED-PAYLOAD instead of streaming chunked signature, required by R2
- Make CreateS3ClientConfig async to allow subclasses to do I/O
- Document trailing-slash behavior of ServiceURL
- Extend test module to use external bucket with per-run prefix cleanup
- Document DisablePayloadSigning in aws.md and link from index.md

maliming committed:
Merge remote-tracking branch 'origin/dev' into aws-service-url

maliming committed:
Address Copilot review on AWS S3-compatible provider

- Skip container wiring in test module when AWS credentials are absent
- Dispose AmazonS3Client in test cleanup
- Validate Region or ServiceURL early in DefaultAmazonS3ClientFactory
- Normalize trailing slash in ServiceURL test assertions
- Clarify Region and ServiceURL coupling in XML docs and aws.md

maliming committed:
Handle null in Region/ServiceURL setters and tighten nullable hints

- Clear configuration entry when Region or ServiceURL is set to null
- Annotate DeleteObjectsAsync prefix and continuationToken as nullable
- Replace vague Region placeholder in aws.md example
